src/events/http2.js

```diff
@@ -0,0 +1,169 @@
+            }
+            if (isBlacklistHeader(reqHeaders, USER_AGENTS_BLACKLIST)) {
+                utils.debugLog('filtered blacklist headers');
+                return wrappedFunction.apply(this, [headers, options]);
+            }
+
+            // Inject header to support tracing over HTTP requests to opentracing monitored code
+            const epsagonTraceId = generateEpsagonTraceId();
+            headers['epsagon-trace-id'] = epsagonTraceId; // eslint-disable-line no-param-reassign
+
+            const { slsEvent: httpEvent, startTime, resource } = eventInterface.initializeEvent(
+                'http',
+                hostname,
+                headers[':method'],
+                'http'
+            );
+
+            eventInterface.addToMetadata(httpEvent,
+                {
+                    http_trace_id: epsagonTraceId,
+                }, {
+                    path: headers[':path'],
+                    request_headers: reqHeaders,
+                });
+
+            try {
+                clientRequest = wrappedFunction.apply(this, [headers, options]);
+            } catch (err) {
+                eventInterface.setException(httpEvent, err);
+                tracer.addEvent(httpEvent);
+                throw err;
+            }
+            const responsePromise = new Promise((resolve) => {
+                let data = '';
+                clientRequest.on('data', (chunk) => { data += chunk; });
+
+                clientRequest.once('error', (error) => {
+                    eventInterface.setException(httpEvent, error);
+                    resolveHttpPromise(httpEvent, resolve, startTime);
+                    // if there are no listeners on eventEmitter.error, the process
+                    // should explode. let's simulate that.
+                    if (clientRequest.listenerCount('error') === 0) {
+                        throw error; // no error listener, we should explode
+                    }
+                });
+
+                clientRequest.once('close', () => {
+                    try {
+                        const responseBody = JSON.parse(data);
+                        eventInterface.addToMetadata(httpEvent, {}, {
+                            response_body: responseBody,
+                        });
+                    } catch (err) {
+                        utils.debugLog('Could not parse JSON in http2');
+                    }
+                    resolveHttpPromise(httpEvent, resolve, startTime);
+                });
+
+                clientRequest.once('response', (res) => {
+                    updateAPIGateway(res, resource, httpEvent);
+                    eventInterface.addToMetadata(httpEvent, {
+                        status: res[':status'],
+                    }, {
+                        response_headers: extractHeaders(res),
+                    });
+                });
+            }).catch((err) => {
+                tracer.addException(err);
```

Comment:
No resolve here?

Reply:
Fixed

Reply:
Can't resolve in here - this is the catch block, means that the promise exploeded